platform.md

```diff
@@ -746,27 +746,31 @@ Usage:
 | `<direct>`          |                       |                | Process execution strategy
 | `<cmd>`             |                       |                | Command to execute
 | `<args>`            |                       |                | Arguments to command
+| `<working-directory>`              |        |                | Process working directory
 | `<layers>/config/metadata.toml`    |        |                | Build metadata (see [`metadata.toml`](#metadatatoml-toml)
 | `<layers>/<buildpack-id>/<layer>/` |        |                | Launch Layers
 
-A command (`<cmd>`), arguments to that command (`<args>`), and an execution strategy (`<direct>`) comprise a process definition. Processes MAY be buildpack-defined or user-defined.
+A command (`<cmd>`), arguments to that command (`<args>`), a working directory (`<working-directory>`), and an execution strategy (`<direct>`) comprise a process definition. Processes MAY be buildpack-defined or user-defined.
 
 The launcher:
-- MUST derive the values of `<cmd>`, `<args>`, and `<direct>` as follows:
+- MUST derive the values of `<cmd>`, `<args>`, `<working-directory>`, and `<direct>` as follows:
 - **If** the final path element in `$0`, matches the type of any buildpack-provided process type
     - `<process-type>` SHALL be the final path element in `$0`
     - The lifecycle:
         - MUST select the process with type equal to `<process-type>` from `<layers>/config/metadata.toml`
         - MUST append any user-provided `<args>` to process arguments
+        - MUST set `<working-directory>` to `<app>` if not defined
 - **Else**
     - **If** `$1` is `--`
         - `<direct>` SHALL be `true`
         - `<cmd>` SHALL be `$2`
         - `<args>` SHALL be `${@3:}`
+        - `<working-directory>` SHALL be `<app>`
```

Comment:
I'm trying to remember all the edge cases here, but is there a reason this just overrides whatever `<working-directory>` is passed in as for these two cases?

Reply:
Jesse just pointed out that this is the direct case, which makes sense.